src/main/java/it/smartcommunitylab/aac/webauthn/service/WebAuthnRpService.java

```diff
@@ -210,23 +222,32 @@ private String generateNewKey() {
                     }
                 }
                 if (toUpdate.isEmpty()) {
-                    return Optional.empty();
+                    throw new WebAuthnAuthenticationException(account.getSubject(),
+                            "Could not find the requested credential in the account");
                 }
                 WebAuthnCredential credential = toUpdate.get();
                 credentials.remove(credential);
                 credential.setSignatureCount(result.getSignatureCount());
                 credential.setLastUsedOn(new Date());
                 webAuthnCredentialsRepository.save(credential);
-                return Optional.of(account.getUsername());
+                final String username = account.getUsername();
+                if (username != null) {
+                    return username;
+                } else {
+                    throw new WebAuthnAuthenticationException(account.getSubject(),
+                            "Could not find the requested credential in the account");
+                }
+
             }
         } catch (Exception e) {
         }
-        return Optional.empty();
+        throw new WebAuthnAuthenticationException("_",
+                "Login failed");
     }
 
-    UserIdentity getUserIdentityOrGenerate(String username, String realm, Optional<String> displayNameOpt,
-            Optional<Subject> optSub) {
-        String displayName = displayNameOpt.orElse("");
+    UserIdentity getUserIdentityOrGenerate(String username, String realm, String displayNameOrNull,
```

Comment:
please name variables without the _Opt,_orNull etc.

Reply:
Done in 35bce9aa2d92f02b01bc8e0f731acdd42de1100c